src/Tribe/iCal.php

```diff
@@ -157,14 +157,89 @@ public function do_ical_template() {
 				$events    = tribe_get_events( array( 'post__in' => $event_ids ) );
 				$this->generate_ical_feed( $events );
 			} elseif ( is_single() ) {
-				$this->generate_ical_feed( $wp_query->post );
+				if ( is_singular( Tribe__Events__Main::POSTTYPE ) ) {
```

Comment:
🤔 in the interests of minimizing the nesting of if blocks, what do you think about shifting from:

```
} elseif ( is_single() ) {
    if ( is_singular( /*...*/ ) {
        # ...
    } else {
        # ...
    }
} else { #...
```

To something like:

```
} elseif ( is_singular( /*...*/ ) {
    # ...
} elseif ( is_single() ) {
    # ...
} else { #...
```

Reply:
Seems more readable in that way ;)